src/Servers/Kestrel/Core/src/KestrelServerOptions.cs

```diff
@@ -36,6 +36,21 @@ public class KestrelServerOptions
 
         private Func<string, Encoding?> _responseHeaderEncodingSelector = DefaultHeaderEncodingSelector;
 
+        private bool? _enableInsecureAbsoluteFormHostOverride;
+        internal bool EnableInsecureAbsoluteFormHostOverride
```

Comment:
Is a public version of this the API proposal for .NET 7?

Reply:
Should we allow for a missing Host header if this is set and there's an absolute form request target?

Reply:
The customer is currently asking for a short term mitigation, it's not clear we'll still need this in 7.0.

> Should we allow for a missing Host header if this is set and there's an absolute form request target?

We could expand to that if needed, but it hasn't come up yet. So far the client is sending a Host header, but it's malformed.

Reply:
I think it's weird not to allow a missing HOST header if we're going to ignore it anyway. What does httpsys do?

Reply:
Http.Sys also always requires the Host header to be present (even if empty), even if it overrides it with the host from the request line.